src/ripple/app/paths/impl/PaySteps.cpp

```diff
@@ -200,9 +202,13 @@ toStrand (
         pes.push_back (&*sendMaxPE);
     for (auto& i : path)
         pes.push_back (&i);
-    auto const lastCurrency =
-        (*boost::find_if (boost::adaptors::reverse (pes), hasCurrency))->getCurrency ();
-    if (lastCurrency != deliver.currency)
+
+    // Note that for offer crossing (only) we do use an offer book even if
+    // all that is changing is the Issue.account.
+    auto const lastSteCurrency =
```

Comment:
Great question.  I couldn't remember either.  It's a pointer to `STPathElement`.  Maybe this is one of those situations where the code would be clearer with an explicit type instead of an `auto`.  Perhaps something like this:
```
STPathElement const* const lastCurrency =
```
Waddya think?